tools/e2e/examples/docker-compose/docker-compose.yaml

```diff
@@ -0,0 +1,195 @@
+version: '3.5'
+
+services:
+
+  #
+  # Kuma Control Plane in "universal" mode with "in-memory" configuration store.
+  #
+  kuma-control-plane:
+    # image name must be provided via environment variable KUMA_CP_DOCKER_IMAGE
+    image: ${KUMA_CP_DOCKER_IMAGE:-kuma/kuma-cp:latest}
+    command:
+    - run
+    - --log-level=info
+    environment:
+    # DNS name of the Kuma xDS server
+    - KUMA_BOOTSTRAP_SERVER_PARAMS_XDS_HOST=kuma-control-plane
+    expose:
+    - "5678"
+    - "5679"
+    - "5680"
+    - "5681"
+    - "5682"
+    ports:
+    - "5678:5678"
+    - "5679:5679"
+    - "5680:5680"
+    - "5681:5681"
+    - "5682:5682"
+    networks:
+      kuma-example:
+        aliases:
+        - kuma-control-plane
+    restart: on-failure
+
+  #
+  # Installation script for Kuma example.
+  #
+  # Uses `kumactl` to create Dataplane resources, generate dataplane tokens, etc.
+  #
+  kuma-example-installer:
+    # image name must be provided via environment variable KUMACTL_DOCKER_IMAGE
+    image: ${KUMACTL_DOCKER_IMAGE:-kuma/kumactl:latest}
+    volumes:
+    - ./kuma-example-installer.sh:/kuma-example/kuma-example-installer.sh
+    command:
+      - /kuma-example/kuma-example-installer.sh
+    networks:
+      kuma-example: {}
+    depends_on:
+      - kuma-control-plane
+    restart: on-failure
+
+  #
+  # Auxiliary service (container) for sharing Linux network namespace
+  # between Example Application and Kuma Sidecar.
+  #
+  kuma-example-app-namespace:
```

Comment:
I don't think I understand this. Can you explain why simple
```
networks:
  kuma-example-app
```
is not enough?

Reply:
Without a shared `Linux network namespace` containers cannot talk to each other over `localhost`.

Personally, I'm not considering merging 2 containers (`application`-specific container and `kuma-dp` container) in one a valid option. It kills the foundational idea of service mesh - ability to evolve your network infrastructure independently from business apps.

Reply:
Yeah, I agree that merging 2 containers is not a best idea here.